### PR TITLE
improve tripperware request id

### PR DESCRIPTION
**Pull request by instabledesign**

### CODE REVIEW

**Review by pimousse1099 — DISMISSED** [review]
maybe add some check on the tests ?

**Comment by pimousse1099** [comment]
rebase needed

**Comment by pimousse1099** [comment]
> rebase needed

done

File: tripperware/request_id_test.go [hunk 1]
```diff
@@ -27,11 +27,13 @@ func TestRequestId(t *testing.T) {
 	roundTripperMock.On("RoundTrip", mock.AnythingOfType("*http.Request")).Return(resp, nil).Run(func(args mock.Arguments) {
 		innerReq := args.Get(0).(*http.Request)
 		assert.True(t, len(innerReq.Header.Get(request_id.HeaderName)) == 10)
+		assert.Equal(t, req.Header.Get(request_id.HeaderName), innerReq.Header.Get(request_id.HeaderName))
 	})
 
 	resp2, err := tripperware.RequestId(request_id.NewConfig())(roundTripperMock).RoundTrip(req)
 	assert.Nil(t, err)
 	assert.Equal(t, resp, resp2)
+	assert.NotEqual(t, req.Header.Get(request_id.HeaderName), "")
```
Comment (instabledesign) [inline tripperware/request_id_test.go]: PRNG
`assert.Equal(t, "p1LGIehp1s", req.Header.Get(request_id.HeaderName))`

**Review by pimousse1099 — DISMISSED** [review]

**Review by qneyrat — DISMISSED** [review]

File: tripperware/request_id_test.go [hunk 2]
```diff
@@ -2,19 +2,28 @@ package tripperware_test
 
 import (
 	"fmt"
+	"github.com/stretchr/testify/assert"
```
Comment (pimousse1099) [inline tripperware/request_id_test.go]: Sort imports

File: tripperware/request_id_test.go [hunk 3]
```diff
@@ -2,19 +2,28 @@ package tripperware_test
...
-	"github.com/stretchr/testify/mock"
-
 	"github.com/gol4ng/httpware"
 	"github.com/gol4ng/httpware/mocks"
 	"github.com/gol4ng/httpware/request_id"
 	"github.com/gol4ng/httpware/tripperware"
 )
 
+func TestMain(m *testing.M){
+	request_id.DefaultIdGenerator = request_id.NewRandomIdGenerator(
```
Comment (pimousse1099) [inline tripperware/request_id_test.go:21]: Nice

**Review by pimousse1099 — CHANGES_REQUESTED** [review]

**Pull request merged**